src/Functions/FunctionFactory.h

```diff
@@ -82,3 +82,31 @@ class FunctionFactory : private boost::noncopyable,
 };
 
 }
+
+typedef void (*func_entry)(DB::FunctionFactory &factory);
+
+#define COLD_INIT __attribute((__section__(".init.text"))) __attribute__((__cold__))
```

Comment:
Maybe something less common? i.e. `FUNCTION_COLD_INIT`, even though it is not specific to functions, but it locales in functions header.